src/FuncSharp/Collections/IEnumerableExtensions_Generic.cs

```diff
@@ -1,18 +1,19 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.Contracts;
 using System.Linq;
 
 namespace FuncSharp
 {
     public static partial class IEnumerableExtensions
     {
         /// <summary>
-        /// Returns a ToList() juts of type IReadOnlyList.
+        /// Returns a ToArray() juts of type IReadOnlyList.
```

Comment:
If you want to return Array:
```suggestion
        /// Uses ToArray to generate an IReadOnlyList.
```

Reply:
My only concern is the potential performance hit with `ToArray` when working with large datasets